### ISSUE TITLE
session_set_save_handler(): Cannot change save handler when session is active

### ISSUE THREAD

**Issue by timm2k** [felixrupp/user_cas#98]
**Describe the bug**
Gettings lots of `session_set_save_handler(): Cannot change save handler when session is active` from `nextcloud/apps/user_cas/vendor/jasig/phpcas/source/CAS/Client.php#1116`

Seems to be a bug in CAS 1.3.9
https://github.com/apereo/phpCAS/issues/365
https://github.com/apereo/phpCAS/issues/372

**To Reproduce**
1. Install user_cas
2. Login via CAS
3. See error in Nextcloud log

**Expected behavior**
No warnings or errors.

**Full error message**
```
Mar 29 04:26:03 ncXY Nextcloud[26064]: {"reqId":"id","level":3,"time":"2021-03-29T04:26:03+02:00","remoteAddr":"123.456.789.101","user":"USER","app":"PHP","method":"PROPFIND","url":"/remote.php/webdav/","message":"{\"Exception\":\"Error\",\"Message\":\"session_set_save_handler(): Cannot change save handler when session is active at /var/www/html/nextcloud/apps/user_cas/vendor/jasig/phpcas/source/CAS/Client.php#1116\",\"Code\":0,\"Trace\":[{\"function\":\"onError\",\"class\":\"OC\\\\Log\\\\ErrorHandler\",\"type\":\"::\",\"args\":[2,\"session_set_save_handler(): Cannot change save handler when session is active\",\"/var/www/html/nextcloud/apps/user_cas/vendor/jasig/phpcas/source/CAS/Client.php\",1116,{\"sessionHandler\":{\"__class__\":\"CAS_Session_PhpSession\"}}]},{\"file\":\"/var/www/html/nextcloud/apps/user_cas/vendor/jasig/phpcas/source/CAS/Client.php\",\"line\":1116,\"function\":\"session_set_save_handler\",\"args\":[{\"__class__\":\"CAS_Session_PhpSession\"},true]},{\"file\":\"/var/www/html/nextcloud/apps/user_cas/vendor/jasig/phpcas/source/CAS/Client.php\",\"line\":937,\"function\":\"setSessionHandler\",\"class\":\"CAS_Client\",\"type\":\"->\",\"args\":[{\"__class__\":\"CAS_Session_PhpSession\"}]},{\"file\":\"/var/www/html/nextcloud/apps/user_cas/vendor/jasig/phpcas/source/CAS.php\",\"line\":365,\"function\":\"__construct\",\"class\":\"CAS_Client\",\"type\":\"->\",\"args\":[\"3.0\",false,\"cas01.elearning.uni-oldenburg.de\",443,\"/cas\",true,{\"__class__\":\"CAS_Session_PhpSession\"}]},{\"file\":\"/var/www/html/nextcloud/apps/user_cas/lib/Service/AppService.php\",\"line\":308,\"function\":\"client\",\"class\":\"phpCAS\",\"type\":\"::\",\"args\":[\"3.0\",\"cas01.elearning.uni-oldenburg.de\",443,\"/cas\"]},{\"file\":\"/var/www/html/nextcloud/apps/user_cas/lib/User/NextBackend.php\",\"line\":132,\"function\":\"init\",\"class\":\"OCA\\\\UserCAS\\\\Service\\\\AppService\",\"type\":\"->\",\"args\":[]},{\"file\":\"/var/www/html/nextcloud/lib/private/User/Manager.php\",\"line\":225,\"function\":\"checkPassword\",\"class\":\"OCA\\\\UserCAS\\\\User\\\\NextBackend\",\"type\":\"->\",\"args\":[\"*** sensitive parameters replaced ***\"]},{\"file\":\"/var/www/html/nextcloud/lib/private/User/Manager.php\",\"line\":202,\"function\":\"checkPasswordNoLogging\",\"class\":\"OC\\\\User\\\\Manager\",\"type\":\"->\",\"args\":[\"*** sensitive parameters replaced ***\"]},{\"file\":\"/var/www/html/nextcloud/lib/private/User/Session.php\",\"line\":755,\"function\":\"checkPassword\",\"class\":\"OC\\\\User\\\\Manager\",\"type\":\"->\",\"args\":[\"*** sensitive parameters replaced ***\"]},{\"file\":\"/var/www/html/nextcloud/lib/private/User/Session.php\",\"line\":789,\"function\":\"checkTokenCredentials\",\"class\":\"OC\\\\User\\\\Session\",\"type\":\"->\",\"args\":[{\"id\":17121559,\"__class__\":\"OC\\\\Authentication\\\\Token\\\\PublicKeyToken\"},\"*** sensitive parameter replaced ***\"]},{\"file\":\"/var/www/html/nextcloud/lib/private/User/Session.php\",\"line\":268,\"function\":\"validateToken\",\"class\":\"OC\\\\User\\\\Session\",\"type\":\"->\",\"args\":[\"*** sensitive parameters replaced ***\"]},{\"file\":\"/var/www/html/nextcloud/lib/private/User/Session.php\",\"line\":243,\"function\":\"validateSession\",\"class\":\"OC\\\\User\\\\Session\",\"type\":\"->\",\"args\":[]},{\"file\":\"/var/www/html/nextcloud/lib/private/legacy/OC_App.php\",\"line\":351,\"function\":\"getUser\",\"class\":\"OC\\\\User\\\\Session\",\"type\":\"->\",\"args\":[]},{\"file\":\"/var/www/html/nextcloud/lib/private/legacy/OC_App.php\",\"line\":114,\"function\":\"getEnabledApps\",\"class\":\"OC_App\",\"type\":\"::\",\"args\":[]},{\"file\":\"/var/www/html/nextcloud/remote.php\",\"line\":152,\"function\":\"loadApps\",\"class\":\"OC_App\",\"type\":\"::\",\"args\":[[\"filesystem\",\"logging\"]]}],\"File\":\"/var/www/html/nextcloud/lib/private/Log/ErrorHandler.php\",\"Line\":91,\"CustomMessage\":\"--\"}","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.14.1","version":"19.0.9.1"}
```

**Software (please complete the following information):**
 - Server-OS: [RHEL 7
 - HTTP-Server Version: nginx 1.18.0
 - PHP-Version: PHP 7.4
 - phpCAS-Library-Version: phpCAS 1.3.9
 - ownCloud/Nextcloud Version: Nextcloud 19.0.9
- user_cas-Version: 1.9.0

**Additional context**
Downgrade phpCAS to 1.3.8 should fix this problem.

**Comment by timm2k** [comment]
*Workaround*
Downgrade to phpCAS 1.3.8

```
cd /var/www/html
wget https://github.com/apereo/phpCAS/releases/download/1.3.8/CAS-1.3.8.tgz
tar xzf CAS-1.3.8.tgz
in NC-Adminpanel -> Security -> CAS Authentication Backend 1.9.0 -> phpCAS Library ->  *Overwrite phpCAS path* with `/var/www/html/CAS-1.3.8/CAS.php`
```

Replace /var/www/html with a path matching your env.

**Comment by landryb** [comment]
to the ones facing this issue and using a phpcas 1.3.9 shipped with user_cas, just apply manually https://github.com/apereo/phpCAS/commit/ec2eda3e5ed926be8e76a002a678133ab6279686 to `apps/user_cas/vendor/jasig/phpcas/source/CAS/Client.php`

**Comment by thomas-mc-work** [comment]
Your mentioned commit is included in the latest version of phpCAS [`1.4.0`](https://github.com/apereo/phpCAS/commits/1.4.0). It would be lovely to have a patch release of user_cas including this dependency version (@felixrupp )! :heart:

**Comment by felixrupp** [comment]
Thanks for your contribution, I just released a bumped version including phpCAS 1.5.0 release.
